addons/controllers/clusterbootstrap_controller.go

```diff
@@ -170,17 +171,22 @@ func (r *ClusterBootstrapReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	log.Info("Reconciling cluster")
 
+	remoteClient, err := util.GetClusterClient(r.context, r.Client, r.Scheme, clusterapiutil.ObjectKey(cluster))
```

Comment:
The reason for moving this `GetClusterClient` up is because the remoteClient can be shared by reconcileDelete, so we'd better call it once and pass the remoteClient to individual functions as param.